### PR TITLE
XGBoost: make gblinear deterministic

### PR DESCRIPTION
**Pull request by honzasterba**
- set updater for gblinear when running on CPU to coord_descent to get rid of indeterminism

### CODE REVIEW

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java [hunk 1]
```diff
@@ -212,6 +212,27 @@ public void dump(String format) {
...
+        return XGBoostParameters.Backend.gpu;
+      } else {
+        Log.info("No GPU (gpu_id: " + p._gpu_id + ") found. Using CPU backend.");
+        return XGBoostParameters.Backend.gpu;
+      }
+    } else {
+      Log.info("Using CPU backend.");
+      return XGBoostParameters.Backend.cpu;
+    }
+  }
```
Comment (honzasterba) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java]: I separated the part of the code that does the backend decision from the part that sets the params based on selected backend and booster

File: h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostPredictImplComparisonTest.java [hunk 2]
```diff
@@ -30,23 +30,20 @@ public static void setup() {
             {"gbtree", "gamma", "AGE"},
             {"gbtree", "poisson", "AGE"},
             {"gbtree", "tweedie", "AGE"},
-            {"gbtree", "gamma", "AGE"},
```
Comment (honzasterba) [inline h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostPredictImplComparisonTest.java]: duplicit param settings removed

**Comment by michalkurka** [comment]
Can we assume that users who don't set the seed explicitly don't care about model reproducibility and resort to the safe version coordinate descent only when the seed is set?

Any deviation from xgboost default parameters should also be documented.

**Comment by honzasterba** [comment]
I dont like the idea of using different updaters with different seed settings. I would not want to document that. Since determinism with same seed is important to us I would stick with coord_descent and document it.

**Pull request merged**